lib/services/core/ads/ads_google.dart

```diff
@@ -0,0 +1,126 @@
+          }, onAdFailedToLoad: (LoadAdError error) async {
+            updateState(placement, AdState.failedLoad, error.toString());
+            placement.nativeAd = null;
+            placement.attempts++;
+            await Future.delayed(waitingDuration);
+            if (placement.attempts <= maxFailedLoadAttempts) {
+              request(type);
+              // } else if (_initialSDK == AdSDKName.google) {
+              //   initialize(args: [AdSDKName.unity]); // Alternative AD SDK
+            }
+          }));
+      return;
+    }
+    RewardedAd.load(
+        adUnitId: getId(AdType.rewarded),
+        request: _request,
+        rewardedAdLoadCallback:
+            RewardedAdLoadCallback(onAdLoaded: (RewardedAd ad) {
+          placement.nativeAd = ad;
+          updateState(placement, AdState.loaded);
+        }, onAdFailedToLoad: (LoadAdError error) async {
+          updateState(placement, AdState.failedLoad, error.toString());
+          placement.data = null;
+          placement.attempts++;
+          await Future.delayed(waitingDuration);
+          if (placement.attempts <= maxFailedLoadAttempts) {
+            request(type);
+            // } else if (_initialSDK == AdSDKName.google) {
+            //   initialize(args: [AdSDKName.unity]); // Alternative AD SDK
+          }
+        }));
+  }
+
+  @override
+  Future<Placement?> show(AdType type, {String? origin}) async {
+    var placement = isReady(AdType.rewarded);
+    if (placement == null) {
+      return null;
+    }
+    if (type.isIntrestitial) {
+      var nativeAd = placement.data as InterstitialAd;
+      nativeAd.fullScreenContentCallback = FullScreenContentCallback(
+          onAdDismissedFullScreenContent: (InterstitialAd ad) =>
+              updateState(placement, AdState.closed),
+          onAdFailedToShowFullScreenContent:
+              (InterstitialAd ad, AdError error) =>
+                  updateState(placement, AdState.failedShow, error.toString()),
+          onAdImpression: (ad) => updateState(placement, AdState.show));
+      nativeAd.show();
+      await waitForClose(type);
+      resetAd(placement);
+      return placement;
+    }
+    // sound.stop("music");
+    var nativeAd = placement.nativeAd as RewardedAd;
+    nativeAd.fullScreenContentCallback = FullScreenContentCallback(
+        onAdDismissedFullScreenContent: (RewardedAd ad) =>
+            updateState(placement, AdState.closed),
+        onAdFailedToShowFullScreenContent: (RewardedAd ad, AdError error) =>
+            updateState(placement, AdState.failedShow, error.toString()),
+        onAdImpression: (ad) => updateState(placement, AdState.show));
+    nativeAd.setImmersiveMode(true);
+    nativeAd.show(
+        onUserEarnedReward: (AdWithoutView ad, RewardItem rewardItem) {
+      placement.reward = rewardItem;
+    });
+    await waitForClose(placement.type);
+    resetAd(placement);
+    // sound.play("african-fun", channel: "music");
```

Comment:
Here we are going to need the `Sound Service`?

Reply:
stop music while ad shown.